src/patrick/whichTwice.js

```diff
@@ -0,0 +1,51 @@
+// whichTwiceRange - (easy-medium)
+//
+// We have an array of n + 1 numbers. Every number in the range 1..n appears once
+// except for one number that appears twice.
+//
+// Write a function for finding the number that appears twice. We can do this
+// in O(n) time and O(1) space! Don't forget the range described above is a
+// triangular series 😉
+//
+// call your function whichTwiceRange()
+
+function whichTwiceRange(arr) {
+  let numbers = [];
+  for (let i = 0; i < arr.length; i++) {
+    if (numbers.length === 0) {
+      numbers.push(arr[i]);
+    } else {
+      if (numbers[numbers.length - 1] === arr[i]) {
+        return arr[i];
+      } else {
+        numbers.push(arr[i]);
+      }
+    }
+  }
+}
```

Comment:
good solution! Accomplishes with worst case O(n) single iteration and only one variable created. Nice!

There's also a quadratic equation solution for ranges of the pattern x, x+1, x+2...n which is how I solved it if you want to go look at my solution, though we both have the same time and space complexity!

🤘